src/interactions/slashCommands/mixins/CommandChannelOptionBase.ts

```diff
@@ -55,10 +55,12 @@ export abstract class ApplicationCommandOptionWithChannelTypesBase
 	}
 
 	public override toJSON(): APIApplicationCommandChannelOptions {
+		// TODO: Fix types
+		// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
 		return {
 			...super.toJSON(),
-			type: ApplicationCommandOptionType.Channel,
+			type: this.type,
 			channel_types: this.channelTypes,
-		};
+		} as APIApplicationCommandChannelOptions;
```

Comment:
Why do you need this cast again?

Reply:
Because the union of all types of command options implies that any field that is required across the entire union is required here. 

In this case we have one member in the union where `autocomplete` is a required field. Therefore the compiler gets mad if it's not provided.

Reply:
That makes no sense.. Is this because of the generic change? If it is, please revert that, as there's no benefits from the generics (as each of the options sets the type anyways)

Reply:
It does, try it out for yourself. 

`type` can be any valid option type. Because of this TS assumes that it's all members of the union. Hence any required field in the union becomes required here. The generics have no bearing here.